make/manifests.mk

```diff
@@ -28,8 +28,16 @@ static-manifests: bin/yaml/cert-manager.crds.yaml bin/yaml/cert-manager.yaml
 ###################
 
 .PHONY: release-manifests
+## Build YAML manifests and helm charts including the helm chart signature
+##
+## Since this command signs artifacts, this requires CMREL_KEY to be configured.
+## Prefer `make release-artifacts` locally.
```

Comment:
Note (unrelated to the contents of the PR): at the moment I think there isn't yet a make workflow for building release artifacts locally for deploying on i.e GKE cluster as we don't have a bazel equivalent for pushing the images

Reply:
I'm not sure I understand correctly, could you rephrase that?

Reply:
It's really absolutely unrelated to this PR's contents, I was just thinking loudly why smo would use this locally as opposed to just running `make-setup..` and whether we want to support that workflow. I usually build artifacts separately if I deploy to GKE, but there is an extra step now needed to get the images to GCR as I think we don't have a make alternative to this https://github.com/cert-manager/cert-manager/blob/release-1.5/Makefile#L143-L152

Reply:
Ahh I'm with you now, yeah. We could totally add something like that target!

The reason I'd want this locally is it would give me confidence that I've not broken the build after making a change. `make release-artifacts-signed` is part of `make release` which in turn is the [literal command we use](https://github.com/cert-manager/release/blob/81115144cf94466e9df5ea70949b0a556e5b0546/gcb/makestage/cloudbuild.yaml#L25) to build cert-manager releases now.

Since most users don't care about the manifest files and most won't be able to generate helm-chart signatures, `make release-artifacts` becomes the closest thing to "build everything possible for a release, just to make sure it all works before I raise a PR".

In short: I think my view of the purpose this particular target is more developer peace-of-mind than for people to regularly use the contents of it!

Reply:
makes sense, thanks for that explanation!

yeah eventually a 'push images' target could be useful I think